examples/dummy_aposmm_libE_gen/run_example.py

```diff
@@ -0,0 +1,99 @@
+"""Basic example of parallel random sampling with simulations."""
+
+from math import gamma, pi, sqrt
+import numpy as np
+from libensemble.generators import APOSMM
+import libensemble.gen_funcs
+
+libensemble.gen_funcs.rc.aposmm_optimizers = "nlopt"
+from optimas.core import VaryingParameter, Objective
+from libensemble.tests.regression_tests.support import (
+    six_hump_camel_minima as minima,
+)
+
+# from optimas.generators import RandomSamplingGenerator
+from optimas.generators import libEWrapper
+from optimas.evaluators import TemplateEvaluator
+from optimas.explorations import Exploration
+
+from multiprocessing import set_start_method
+
+set_start_method("fork", force=True)
+
+
+def analyze_simulation(simulation_directory, output_params):
+    """Analyze the simulation output.
+
+    This method analyzes the output generated by the simulation to
+    obtain the value of the optimization objective and other analyzed
+    parameters, if specified. The value of these parameters has to be
+    given to the `output_params` dictionary.
+
+    Parameters
+    ----------
+    simulation_directory : str
+        Path to the simulation folder where the output was generated.
+    output_params : dict
+        Dictionary where the value of the objectives and analyzed parameters
+        will be stored. There is one entry per parameter, where the key
+        is the name of the parameter given by the user.
+
+    Returns
+    -------
+    dict
+        The `output_params` dictionary with the results from the analysis.
+
+    """
+    # Read back result from file
+    with open("result.txt") as f:
+        result = float(f.read())
+    # Fill in output parameters.
+    output_params["f"] = result
+    return output_params
+
+
+# Create varying parameters and objectives.
+var_1 = VaryingParameter("x0", -3.0, 3.0)
+var_2 = VaryingParameter("x1", -2.0, 2.0)
+obj = Objective("f")
+
+n = 2
+
+aposmm = APOSMM(
+    initial_sample_size=100,
+    localopt_method="LN_BOBYQA",
+    sample_points=np.round(minima, 1),
+    rk_const=0.5 * ((gamma(1 + (n / 2)) * 5) ** (1 / n)) / sqrt(pi),
+    xtol_abs=1e-2,
+    ftol_abs=1e-2,
+    dist_to_bound_multiple=0.5,
+    max_active_runs=4,  # refers to APOSMM's simul local optimization runs
+    lb=np.array([-3, -2]),  # potentially matches the VaryingParameters
+    ub=np.array([3, 2]),
+)
+
+gen = libEWrapper(
+    varying_parameters=[var_1, var_2],
+    objectives=[obj],
+    libe_gen_instance=aposmm,
```

Comment:
How different to libe_gen. This can be set up so if a class is provided it initializes in place.
e.g.,
if inspect.isclass(generator):
   initialize...

Reply:
That's true, for rough-development I was being explicit on what type of object we passed, but we can afford to be smarter now. Pass in something initialized, keep as-is, if its not initialized, initialize it.

I'll take a second look, but I don't think I came across a pure-optimas example that did object-initialization within the library instead of exposed to the user.